File: 965-univalued-binary-tree/965-univalued-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool isUnivalTree(TreeNode* root) {

        if (!root)
            return {};

        std::queue<TreeNode *> myQ;
        myQ.push(root);
        
        int rootVal = root->val;

        while (myQ.empty() != true)
        {
            TreeNode *node = myQ.front();
            myQ.pop();
            
            if( node->val != rootVal )
                return false;
            
            if(node->left)
                myQ.push(node->left);
            if(node->right)
                myQ.push(node->right);
        }

        return true;
    }
};

File: 965-univalued-binary-tree/965-univalued-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def isUnivalTree(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        if root is None:
            return true
        
        uniVal = root.val
        q = collections.deque([root])
        
        while len(q) != 0:
            numNode = len(q)
            
            for _ in range(numNode):
                node = q.popleft()
                if node.val != uniVal:
                    return False
                if node.left is not None:
                    q.append(node.left)
                if node.right is not None:
                    q.append(node.right)
        return True